meteor/client/ui/SegmentTimeline/SegmentTimelineContainer.tsx

```diff
@@ -508,38 +508,46 @@ export const SegmentTimelineContainer = translateWithTracker<IProps, IState, ITr
 			}
 			// segment is stopping from being live
 			if (this.state.isLiveSegment === true && isLiveSegment === false) {
-				this.setState({ isLiveSegment: false })
+				this.setState({ isLiveSegment: false }, () => {
+					if (Settings.autoRewindLeavingSegment) {
+						this.onRewindSegment()
+						this.onShowEntireSegment()
+					}
+				})
 				this.stopLive()
-				if (Settings.autoRewindLeavingSegment) {
-					this.onRewindSegment()
-					this.onShowEntireSegment()
-				}
 			}
+
+			// Setting the correct scroll position on parts when setting is next
+			const nextPartDisplayStartsAt =
+				currentNextPart &&
+				this.context.durations?.partDisplayStartsAt &&
+				this.context.durations.partDisplayStartsAt[unprotectString(currentNextPart.partId)]
+			const partOffset =
+				nextPartDisplayStartsAt -
+					this.context.durations.partDisplayStartsAt[unprotectString(this.props.parts[0].instance.part._id)] || 0
+			const nextPartIdOrOffsetHasChanged =
+				currentNextPart &&
+				this.props.playlist.nextPartInstanceId &&
+				(prevProps.playlist.nextPartInstanceId !== this.props.playlist.nextPartInstanceId ||
+					this.nextPartOffset !== partOffset)
+			const isBecomingNextSegment = this.state.isNextSegment === false
```

Comment:
Shouldn't this be `const isBecomingNextSegment = this.state.isNextSegment === false && isNextSegment`?
Also, please keep the comment in for the ifs, it makes it much easier to return to this code later and understand what each of it's parts is supposed to do. I know that the variable names are somewhat self-explanatory, but still.

Reply:
I agree. The reason for not having it was that `isNextSegment` is included in the if-statement. But I will change it to include it in `const isBecomingNextSegment` for better readability.
Just to be on the same page, you want the comment, that describes the use case for the condition, to be placed as the first line inside the condition body?

Reply:
The comment being _"the segment isn't live, is next, and the nextPartId has changed"_, which I guess should now be _"the segment isn't live, will be next, and either the nextPartId has changed or it is just becoming next"_, which I guess maybe points to the fact that some of these checks in that if statement are now redundant? Hmm... 🤔